pkgs/os-specific/linux/kernel/sunlight-kernels.nix

```diff
@@ -0,0 +1,69 @@
+{ lib, stdenv, fetchFromGitHub, buildLinux, ... } @ args:
+
+let
+  ltsVariant = {
+    version = "6.1.15";
+    hash = "sha256-XUH7fxj+t230el/vYVlW/3KtY4E1wQKp1TYT26J7sdg";
+    variant = "lts";
+  };
+
+  sunlightKernelFor = { version, suffix ? "sunlight1", hash, variant }: buildLinux (args // rec {
+    inherit version;
+    modDirVersion = lib.versions.pad 3 "${version}-${suffix}";
+
+    src = fetchFromGitHub {
+      owner = "sunlightlinux";
+      repo = "linux-sunlight";
+      rev = "26395413f0b1b49a21776d1e3804b17d4629db96";
+      inherit hash;
+    };
+
+    kernelPatches = [];
+
+    structuredExtraConfig = with lib.kernel; {
+      GKI_HACKS_TO_FIX = yes;
+
+      # AMD P-state driver
+      X86_AMD_PSTATE = lib.mkOverride 60 yes;
+      X86_AMD_PSTATE_UT = no;
+
+      # Google's BBRv2 TCP congestion Control
+      TCP_CONG_BBR2 = yes;
+      DEFAULT_BBR2 = yes;
+
+      # FQ-PIE Packet Scheduling
+      NET_SCH_DEFAULT = yes;
+      DEFAULT_FQ_PIE = yes;
+
+      # Futex WAIT_MULTIPLE implementation for Wine / Proton Fsync.
+      FUTEX = yes;
+      FUTEX_PI = yes;
+
+      # WineSync driver for fast kernel-backed Wine
+      WINESYNC = module;
+
+      # Preemptive Full Tickless Kernel at 833Hz
+      LATENCYTOP = yes;
+
+      PREEMPT = lib.mkOverride 60 yes;
+      PREEMPT_VOLUNTARY = lib.mkForce no;
+      PREEMPT_NONE = lib.mkForce no;
+
+      HZ = freeform "833";
```

Comment:
This is oddly specific, where does this value come from?

Reply:
833 Hz is alternative to 1000 Hz. I choose this value for a balance between latency, performance and low power consumption.

Reply:
Ideally explain such changes in a comment right in the code.

Reply:
We brought explain with comment right in the code.